pkg/provider/provider.go

```diff
@@ -1432,6 +1475,10 @@ func (k *kubeProvider) Update(
 	// - [ ] Support server-side apply, when it comes out.
 	//
 
+	if k.clusterUnreachable {
+		return nil, fmt.Errorf("configured Kubernetes cluster is unreachable")
+	}
```

Comment:
Minor - Should this go below the logging about executing an `Update`?  I think it would be surprising not to see that `Update` was called.

Reply:
Also - perhaps a comment on why erroring early is correct here?